lib/NormalModuleFactory.js

```diff
@@ -849,6 +849,25 @@ ${err2.stack}`;
 								err.message += `
 ${hints.join("\n\n")}`;
 							}
+
+							// Check if the extension is missing a leading dot (e.g. "js" instead of ".js")
+							let appendResolveExtensionsHint = false;
+							const specifiedExtensions = Array.from(
+								resolver.options.extensions
+							);
+							const expectedExtensions = specifiedExtensions.map(extension => {
+								if (extension.match(/^[^.]/)) {
```

Comment:
Also use `RegExp.prototype.test` instead of `String.prototype.match` since `.match` is stateful and we don't need that state, rather just the boolean return on if a match occurs.